src/board_builder/test/accuracy/accuracy_test.py

```diff
@@ -0,0 +1,224 @@
+        return rms_error
+
+    def run_accuracy_tester(self):
+        module = importlib.import_module(f"src.board_builder.test.accuracy.board_definitions.{self._parameters.SCENE_NAME}")
+        board_definition = getattr(module, "BOARD_DEFINITION")
+
+        ### REFERENCE DATA ###
+        for pose in self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE:
+            self.board_builder.pose_solver.set_intrinsic_parameters(pose.target_id, self._parameters.DETECTOR_INTRINSICS)
+
+        self.board_builder.pose_solver.set_detector_poses(self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE)
+
+        ### GENERATE SNAPSHOTS OF THE SCENE ###
+        snapshots = generate_virtual_snapshots(board_definition, self._parameters.NUMBER_OF_SNAPSHOTS)
+        average_difference = self._calculate_generate_snapshots_inaccuracy(board_definition, snapshots)
+        print(f"Generated snapshots inaccuracy (average Frobenius norm difference): {average_difference}")
+
+        ### COLLECTION DATA ###
+        two_dimension_collection_data = []
+        predicted_corners = []
+
+        for snapshot in snapshots:
+            collection_data = generate_data(snapshot, self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE, remove_markers_out_of_frame=True)
+            two_dimension_collection_data.append(collection_data)
+            noisy_collection_data = self._add_noise_to_corners(collection_data)
+            collection_corners = self.board_builder.collect_data(noisy_collection_data)
+            predicted_corners.append(collection_corners)
+        #graph_renderer(snapshots, two_dimension_collection_data, self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE)
+
+        ### BUILD BOARD ###
+        predicted_board = self.board_builder.build_board(repeatability_testing=False)
+
+        # Center the board definition around the reference marker
+        board_definition_np = {k: np.array(v) for k, v in board_definition.items()}
+        marker_0_corners = board_definition_np[0]
+        center_marker_0 = np.mean(marker_0_corners, axis=0)
+        simulated_board_definition = {str(k): v - center_marker_0 for k, v in board_definition_np.items()}
+
+        simulated_markers = [
+            Marker(marker_id=k, points=v.tolist())
+            for k, v in simulated_board_definition.items()
+        ]
+        simulated_board = TargetBoard(target_id='simulated_board', markers=simulated_markers)
+
+        # RMS
+        # TODO: FIX RMS ERROR
+        #  If we compare predicted board vs simulated board, the y and z value error is lower than 0.5, but the x value
+        #  seems to be consistently off by around 6 mm, which is where the error is mainly coming from. I can't find
+        #  why that is right now.
+        aligned_board = self.align_boards(predicted_board, simulated_board)
+        rms_error = self._calculate_rms_error_of_two_corner_dataset(aligned_board, simulated_board)
+        print(f"RMS Error of board corners: {rms_error}")
```

Comment:
Can you easily output the data (generated boards, frames, output boards, RMS error) to a file? I think it is important to keep that data on hand since we intend to publish it.

Reply:
Just implemented it. I store the following data: parameters, generated_board_poses, proejcted_2D_points, predicted_board, reference_board, RMS. Also, The file is rewritten each run instead of being appended to previous run.